taichi/rhi/interop/vulkan_cpu_interop.cpp

```diff
@@ -15,6 +15,21 @@ namespace lang {
 using namespace taichi::lang::vulkan;
 using namespace taichi::lang::cpu;
 
+void memcpy_cpu_to_vulkan(DevicePtr dst, DevicePtr src, uint64_t size) {
+  VulkanDevice *vk_dev = dynamic_cast<VulkanDevice *>(dst.device);
+  CpuDevice *cpu_dev = dynamic_cast<CpuDevice *>(src.device);
+
+  DeviceAllocation src_alloc(src);
+
+  CpuDevice::AllocInfo src_alloc_info = cpu_dev->get_alloc_info(src_alloc);
+
+  unsigned char *dst_ptr = (unsigned char *)(vk_dev->map_range(dst, size));
+  unsigned char *src_ptr = (unsigned char *)src_alloc_info.ptr + src.offset;
+
+  memcpy(dst_ptr, src_ptr, size);
```

Comment:
Not the best idea... Writing to `dst` here might be very slow because it's on-device memory, or it would be slow for GPU to use. The best solution is to allocate a vulkan buffer that is on the CPU and use that as staging buffer

Reply:
Yes, I know that. However here I just want a function to directly passing cpu data to storage buffer which belongs to transfer family and don't need staging buffer cause it is staging buffer. Maybe I should add a annotation that here the `dst` must a staging buffer (host visible buffer) instead of device buffer.

Reply:
This is not the best API... If `memcpy_direct` requires `dst` to be of specific attribute, it needs proper checking & validation

Reply:
Ok, so what is the best way of copying from cpu data to the ssbo? And I don't see checking and validation in other place. What can I do to cover this?